lib/rules/array-element-newline.js

```diff
@@ -0,0 +1,244 @@
+                    }
+
+                    return fixer.replaceTextRange([twoTokensBefore.range[1], tokenBefore.range[0]], "");
+
+                }
+            });
+        }
+
+        /**
+        * Reports that there should be a line break after the first token
+        * @param {ASTNode} node - The node to report in the event of an error.
+        * @param {Token} token - The token to use for the report.
+        * @returns {void}
+        */
+        function reportRequiredLineBreak(node, token) {
+            let tokenBefore = sourceCode.getTokenBefore(token, { includeComments: true });
+
+            // in case array elements are surrounded by parentheses
+            // the element token does usually not include them.
+            // set token and tokenBefore before these parentheses
+            if (tokenBefore.value === "(") {
+                token = tokenBefore;
+                tokenBefore = sourceCode.getTokenBefore(token, { includeComments: true });
+            }
+
+            context.report({
+                node,
+                loc: {
+                    start: tokenBefore.loc.end,
+                    end: token.loc.start
+                },
+                message: "There should be a linebreak after this element.",
+                fix(fixer) {
+                    return fixer.replaceTextRange([tokenBefore.range[1], token.range[0]], "\n");
+                }
+            });
+        }
+
+        /**
+         * Reports a given node if it violated this rule.
+         *
+         * @param {ASTNode} node - A node to check. This is an ObjectExpression node or an ObjectPattern node.
+         * @param {{multiline: boolean, minItems: number}} options - An option object.
+         * @returns {void}
+         */
+        function check(node) {
+            const elements = node.elements;
+            const normalizedOptions = normalizeOptions(context.options[0]);
+            const options = normalizedOptions[node.type];
+
+            let elementBreak = false;
+
+            /*
+             * MULTILINE: true
+             * loop through every element and check
+             * if at least one element has linebreaks inside
+             * this ensures that following is not valid (due to elements are on the same line):
+             *
+             * [
+             *      1,
+             *      2,
+             *      3
+             * ]
+             */
+            if (options.multiline) {
+                elementBreak = elements
+                    .filter(element => element !== null)
+                    .some(element => element.loc.start.line !== element.loc.end.line);
+            }
+
+            const needsLinebreaks = (
+                elements.length >= options.minItems ||
+                (
+                    options.multiline &&
+                    elementBreak
+                )
+            );
+
+            elements.forEach((element, i) => {
+                const lastElement = elements[i - 1];
```

Comment:
Nitpick: Can you rename this variable to `previousElement`? At first, I misread it and thought it was referring to the final element in the array.